### PR TITLE
BaSP-MR-117: Some functionalities fixed and refactoring some parts of the …

### PR DESCRIPTION
**Pull request by YhanahiRosenthal**

### CODE REVIEW

File: src/Components/Activities/AddActivity/Index.jsx [hunk 1]
```diff
@@ -23,17 +23,21 @@ const ModalAddActivity = ({
...
     });
 
-    const allFieldsValid = Object.values(bodyActivity).every((field) => field.length >= 3);
-
     setActive(!allFieldsValid);
   };
 
+  useEffect(() => {
+    setActive(!allFieldsValid);
+  }, [bodyActivity]);
```
Comment (FrancoLelli) [inline src/Components/Activities/AddActivity/Index.jsx:39]:
```suggestion
  }, []);
```

**Review by FrancoLelli — CHANGES_REQUESTED** [review]
You shouldn't make useEffect execute. You could use setActivity and add to the activities array, the new object

File: src/Components/Activities/AddActivity/Index.jsx (hunk 1 above)
Comment (YhanahiRosenthal) [reply src/Components/Activities/AddActivity/Index.jsx]: Already fixed Lord

**Review by FrancoLelli — APPROVED** [review]
Nice Job!

**Review by CristianLotorto — APPROVED** [review]

**Pull request merged**